services/subnet_manager/src/main/java/com/futurewei/alcor/subnet/controller/SubnetController.java

```diff
@@ -449,7 +449,7 @@ public ResponseId deleteSubnetState(@PathVariable String projectId, @PathVariabl
                 return new ResponseId();
             }
 
-            //RestPreconditionsUtil.verifyParameterEqual(subnetEntity.getProjectId(), projectId);
+            this.subnetDatabaseService.deleteSubnet(subnetId);
```

Comment:
In order to grant deletion, we will need to make sure the following two conditions are met:
1. No ports are under this subnet
1. The subnet is not attached to any router

Reply:
sure

Reply:
Very good improvement and thanks.